glusterd2/events/nfs-ganesha.go

```diff
@@ -0,0 +1,37 @@
+package events
+
+import (
+	log "github.com/sirupsen/logrus"
+	"fmt"
+	"os/exec"
+)
+
+
+type ganesha struct{}
+
+func (g *ganesha) Handle(e *Event) {
+		var option string
+		if (e.Name == "volume.started") {
+			option = "on"
+		} else if (e.Name == "volume.stopped") {
+			option = "off"
+		}
+		dbuscmdStr := fmt.Sprintf("/usr/libexec/ganesha/dbus-send.sh /etc/ganesha/ %s %s", option, e.Data["volume.name"])
+		ganeshaCmd := exec.Command("/bin/sh", "-c", dbuscmdStr)
+		err := ganeshaCmd.Run()
+		if err != nil {
+			log.Warn(dbuscmdStr, " got failed with ", err)
+		} else {
+			log.Debug(dbuscmdStr, " got succeed")
+		}
+}
+
+func (g *ganesha) Events() []string {
+	        return []string{"volume.started", "volume.stopped"}
```

Comment:
As these are used at multiple places in the same file, you can simply define them as constants instead:

```go
const (
    eventVolumeStarted = "volume.started"
    eventVolumeStopped = "volume.stopped"
)
```
You can reuse the constants in `Events()` and in `Handle()`

In the future, we may have these events exported from a common package.

Reply:
Done